oneview/resource_server_profile_template.go

```diff
@@ -2482,13 +2487,14 @@ func resourceServerProfileTemplateUpdate(d *schema.ResourceData, meta interface{
 		val := d.Get("hide_unused_flex_nics")
 		serverProfileTemplate.HideUnusedFlexNics = val.(bool)
 	}
-
-	val := d.Get("enclosure_group")
-	enclosureGroup, err := config.ovClient.GetEnclosureGroupByName(val.(string))
-	if err != nil {
-		return err
+	if d.Get("enclosure_group") != "" {
+		val := d.Get("enclosure_group")
+		enclosureGroup, err := config.ovClient.GetEnclosureGroupByName(val.(string))
+		if err != nil {
+			return err
+		}
+		serverProfileTemplate.EnclosureGroupURI = enclosureGroup.URI
 	}
-	serverProfileTemplate.EnclosureGroupURI = enclosureGroup.URI
```

Comment:
i think this line should be inside if condition as done above

Reply:
hey, its inside if block only